crates/matrix-sdk-base/src/store/sled_store/mod.rs

```diff
@@ -168,6 +196,67 @@ pub fn decode_key_value(key: &[u8], position: usize) -> Option<String> {
     values.get(position).map(|s| String::from_utf8_lossy(s).to_string())
 }
 
+#[derive(Clone, Copy, Default, PartialEq)]
+struct BatchIdx(usize);
+
+impl From<sled::IVec> for BatchIdx {
+    fn from(item: sled::IVec) -> Self {
+        Self(usize::from_be_bytes(
+            item.as_ref().try_into().expect("The batch index wasn't properly encoded"),
+        ))
+    }
+}
+
+impl BatchIdx {
+    fn next(self) -> Self {
+        Self(self.0 + 1)
+    }
+
+    fn to_be_bytes(self) -> [u8; 8] {
+        self.0.to_be_bytes()
```

Comment:
This only works on 64-bit architectures. Since `self.0` here is a `usize`, the `[u8; 8]` return type will be invalid as `self.0.to_be_bytes()` returns `[u8; 4]` on 32-bit architectures.